crates/sui-core/src/authority.rs

```diff
@@ -3084,18 +3084,20 @@ impl AuthorityState {
         epoch_start_configuration: EpochStartConfiguration,
     ) -> SuiResult<Arc<AuthorityPerEpochStore>> {
         let new_epoch = new_committee.epoch;
-        info!(new_epoch = ?new_committee.epoch, "re-opening AuthorityEpochTables for new epoch");
+        info!(new_epoch = ?new_epoch, "re-opening AuthorityEpochTables for new epoch");
         assert_eq!(
             epoch_start_configuration.system_state.epoch(),
             new_committee.epoch
         );
+        self.db()
```

Comment:
@lxfind is this safe?
It means that if node crash here we will start a node with new epoch, but without epoch store created. Will we then create epoch store on startup? Is this something we could write unit test for?

Reply:
I convinced myself that multiple invocations of `AuthorityPerEpochStore::new()` are idempotent because it just initializes internal+db state based off of passed in parameters which is all based on epoch number. Having already updated the epoch number, even if the node crashes here - we will just create the new epoch store on startup.

Reply:
It's probably safe, but I am not 100% sure. Agreed a test would be ideal